vault/identity_store_oidc.go

```diff
@@ -415,6 +522,145 @@ func (i *IdentityStore) handleOIDCGenerateSignToken(ctx context.Context, req *lo
 	}, nil
 }
 
+// handleOIDCCreateRole is used to create a new role or update an existing one
+func (i *IdentityStore) handleOIDCCreateRole(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
```

Comment:
This looks functionally correct but I think it can be simplified somewhat so there isn't as much parallel logic for update/!update. Can chat when you're back.